nntrainer/models/neuralnet.h

```diff
@@ -533,7 +525,8 @@ class NeuralNetwork : public ml::train::Model {
 
   AppContext app_context; /** Configurations bound to current app */
 
-  NetworkGraph model_graph; /** Network Model Graph */
+  NetworkGraph model_graph;                 /** Network Model Graph */
+  GraphRepresentation graph_representation; /** Unsorted graph representation */
```

Comment:
this is fine for now, but it is best to maintain the ordering of the nodes given in the definition.
Topological sort ensures that in two sibling nodes, they are executed in the order of the listing. 
The same conditions need to be satisfied to the realizers, storing connections and other graph operations, and then neuralnetwork need not store the graph in two formats.

Reply:
Yeah. This will be likely std::variant<GraphRepresentation, NetworkGraph> graph